client/dive-common/components/AutosavePrompt.vue

```diff
@@ -0,0 +1,52 @@
+<script lang="ts">
+import { defineComponent } from '@vue/composition-api';
+
+export default defineComponent({
+  name: 'AutosavePrompt',
+  props: {
+    value: {
+      type: Boolean,
+      required: true,
+    },
+  },
+});
+</script>
+<template>
+  <v-dialog
+    max-width="500"
+    persistent
+    :value="value"
+    :overlay-opacity="0.95"
+  >
+    <v-card outlined>
+      <v-card-text class="py-4">
+        <p class="text-h5">
+          Do you want to save changes first?
+        </p>
+        <v-alert
+          dense
+          outlined
+          type="warning"
+        >
+          There are unsaved changes to this dataset.
+        </v-alert>
+      </v-card-text>
+      <v-card-actions>
+        <v-spacer />
+        <v-btn
+          depressed
+          text
+          @click="$emit('input', false)"
+        >
+          Cancel
+        </v-btn>
+        <v-btn
+          color="primary"
+          @click="$emit('save')"
+        >
+          Save and Export
+        </v-btn>
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+</template>
```

Comment:
Might hate me for suggesting this.  I really like the inline async-ness and simplicity of the $prompt component and the 'prompt-> set flag->do one of multiple things' cycle is something that is done a lot.  Should we invest in making a compositionAPI tool for this?

Reply:
Yes.  It's used in 7 components.  

Can we make this a separate issue thought?  I think it's worth doing, but not as part of this.

Reply:
Yeah, I think it would be a separate thing.

Reply:
We should get rid of `$snackbar` as well.  It's only used in once place, which can be replaced with just a regular `v-snackbar`